.changes/f71f083b-6e5f-4a3c-9069-464b1f9f6d36.json

```diff
@@ -0,0 +1,5 @@
+{
+    "id": "f71f083b-6e5f-4a3c-9069-464b1f9f6d36",
+    "type": "bugfix",
+    "description": "Fix paginator generator `List<Document>` nullability"
```

Comment:
nit/clarification: This change does not only affect `List<Document>`, right?

Reply:
It _should_ only affect `List<Document>` types because they're the only type of list that can contain nullables without the `sparse` trait being set on the list. Looking at [KotlinSymbolProvider](https://github.com/smithy-lang/smithy-kotlin/blob/main/codegen/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/core/KotlinSymbolProvider.kt), document is the only symbol that always returns `asNullable()`